users/service.go

```
@@ -555,8 +566,12 @@ func (svc service) OAuthCallback(ctx context.Context, user User) (User, error) {
 	u, err := svc.users.RetrieveByEmail(ctx, user.Email)
 
 	if errors.Contains(err, repoerr.ErrNotFound) {
+		user.Credentials.Username = generateUsername(user.Email)
 		u, err = svc.Register(ctx, authn.Session{}, user, true)
 		if err != nil {
+			if errors.Contains(err, errors.ErrUsernameNotAvailable) {
+				return User{}, errors.ErrTryAgain
+			}
```

This will result in losing the proper error in logs. TryAgain should be returned by the API layer, but service layer should not mask the actual error or we will be blind when such an error happens.